http/handler_collection.go

```diff
@@ -0,0 +1,343 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi/v5"
+
+	"github.com/sourcenetwork/defradb/client"
+)
+
+type collectionHandler struct{}
+
+type CollectionDeleteRequest struct {
+	Key    string   `json:"key"`
+	Keys   []string `json:"keys"`
+	Filter any      `json:"filter"`
+}
+
+type CollectionUpdateRequest struct {
+	Key     string   `json:"key"`
+	Keys    []string `json:"keys"`
+	Filter  any      `json:"filter"`
+	Updater string   `json:"updater"`
+}
+
+func (s *collectionHandler) Create(rw http.ResponseWriter, req *http.Request) {
+	col := req.Context().Value(colContextKey).(client.Collection)
+
+	var body any
+	if err := requestJSON(req, &body); err != nil {
+		responseJSON(rw, http.StatusBadRequest, errorResponse{err})
+		return
+	}
+
+	switch t := body.(type) {
```

Comment:
suggestion:  I found the split here to Create/CreateMany strange, and I had to check few a few files to be sure where and why it was like this.  I understand it in the short term, but still think it is strange, and think you are kind of breaking the `client.Collection` interface here - the `http.Client` type implements the interface, but the http endpoint does not - this will be visible to users when not using the Go http-client wrapper.

I suggest you add a new ticket to change the `client.Collection` interface to remove the `CreateMany` func.  Create would then need to handle 0..n docs - I would leave it to whoever picks up that ticket how to handle that (could be an `any` type, `...Document`, or a generic constrait even).

In the new ticket please reference this code area to ensure it gets cleaned up along the way.

Reply:
I had made this trade-off to keep the API more RESTful.

I do like the idea of merging the `Create` and `CreateMany` methods.

Reply:
> I had made this trade-off to keep the API more RESTful.

I understood this :) Please don't forget to create that ticket though.

Reply:
Created issue #1844

Reply:
Thanks :)